### ISSUE TITLE
App Engine SDK version 1.9.62 onwards causes non-persistence of information

### ISSUE THREAD

**Issue by rojakcoder** [gorilla/sessions#132]
Not sure if this is an issue that requires a fix in Gorilla Sessions but I just want to report this in case anyone else has the same problem.

When a Go test simulates log-in by creating a session value, and then serves the request using Gorilla Mux, the value stored in the session is lost to the request handler i.e. the `session.Values` map is empty.

As the error occurs when the SDK is changed, I'm not sure if this is a problem with Sessions or just the App Engine SDK.

**Comment by elithrar** [comment]
Hi @rojakcoder - did you solve this? If not, can you share the version of the App Engine SDK / Go version you're using?

**Comment by rojakcoder** [comment]
Hi @elithrar - I have not got around to solving the problem. I started having the problem with Go 1.8.5 and App Engine SDK 1.9.62.

The App Engine SDK is currently 1.9.65 and I still have the same problem.

**Comment by rojakcoder** [comment]
I've gotten some headways after some debugging.

The session is stored with the registry. The registry is associated with the request (in context.go `map[*http.Request]map[interface{}]interface{}`)

I then proceeded to check the request.

From my tests, I created the session value in cookies to simulate signing in (`session, err := store.Get(req, SessLogin)`).

With go version 1.8.5 (appengine-1.9.65) linux/amd64, the middleware that checks for the user credentials fail. I found out that the `req` variable has a different pointer address in the tests than the one the middleware receives.

When I used go version 1.6.4 (appengine-1.9.54) linux/amd64, the `req` variable has the same pointer address in the test and the middleware layer.

I think that the different pointer address is causing the tests to fail. Any idea why this happens and how to resolve it?
